tests/rgen/Microsoft.Macios.Generator.Tests/Extensions/NamedTypeSymbolExtensionsTests.cs

```diff
@@ -0,0 +1,172 @@
+namespace Test;
+public class NotEnum {
+}
+";
+		var (compilation, syntaxTrees) = CreateCompilation (nameof (TryGetEnumFieldsNotEnum), platform, inputString);
+		Assert.Single (syntaxTrees);
+		var declaration = syntaxTrees [0].GetRoot ()
+			.DescendantNodes ()
+			.OfType<BaseTypeDeclarationSyntax> ()
+			.FirstOrDefault ();
+		Assert.NotNull (declaration);
+		var semanticModel = compilation.GetSemanticModel (syntaxTrees [0]);
+		var symbol = semanticModel.GetDeclaredSymbol (declaration);
+		Assert.NotNull (symbol);
+		Assert.False (symbol.TryGetEnumFields (out var fields, out var diagnostics));
+		Assert.Null (fields);
+		Assert.Single (diagnostics);
+	}
+
+	class TestDataTryGetEnumFieldsNoFields : IEnumerable<object []> {
+		public IEnumerator<object []> GetEnumerator ()
+		{
+
+			foreach (var platform in Configuration.GetIncludedPlatforms (true)) {
+
+				const string emptyEnum = @"
+namespace Test;
+public enum MyEnum {
+}
+";
+				yield return [platform, emptyEnum];
+
+				const string missingFieldAttributes = @"
+namespace Test;
+public enum MyEnum {
+	First,
+	Second,
+	Last,
+}
+";
+				yield return [platform, missingFieldAttributes];
+
+
+				const string fieldWithQuotes = @"
+namespace Test;
+public enum MyEnum {
+	[Field<EnumValue> (""Field\""With\""Quotes"")]
+	First,
+}
+";
+				yield return [platform, fieldWithQuotes];
+
+				const string leadingWithNumber = @"
+namespace Test;
+public enum MyEnum {
+	[Field<EnumValue> (""42Tries"")]
+	First,
+}
+";
+				yield return [platform, leadingWithNumber];
+
+				const string fieldWithNewLines = @"
+namespace Test;
+public enum MyEnum {
+	[Field<EnumValue> (""With\nNew\nLine"")]
+	First,
+}
+";
+				yield return [platform, fieldWithNewLines];
+
+				const string fieldWithKeyword = @"
+namespace Test;
+public enum MyEnum {
+	[Field<EnumValue> (""class"")]
+	First,
+}
+";
+				yield return [platform, fieldWithKeyword];
+
+				const string fieldWithTabs = @"
+namespace Test;
+public enum MyEnum {
+	[Field<EnumValue> ("" \tSecondBackendField\t \n"")]
+	First,
+}
+";
+
+				yield return [platform, fieldWithTabs];
+			}
+		}
+
+		IEnumerator IEnumerable.GetEnumerator () => GetEnumerator ();
+	}
+
+
+	[Theory]
+	[ClassData (typeof (TestDataTryGetEnumFieldsNoFields))]
+	public void TryGetEnumFieldsNoFields (ApplePlatform platform, string inputString)
+	{
+		var (compilation, syntaxTrees) = CreateCompilation (nameof (TryGetEnumFieldsNotEnum), platform, inputString);
+		Assert.Single (syntaxTrees);
+		var declaration = syntaxTrees [0].GetRoot ()
+			.DescendantNodes ()
+			.OfType<BaseTypeDeclarationSyntax> ()
+			.FirstOrDefault ();
+		Assert.NotNull (declaration);
+		var semanticModel = compilation.GetSemanticModel (syntaxTrees [0]);
+		var symbol = semanticModel.GetDeclaredSymbol (declaration);
+		Assert.NotNull (symbol);
+		Assert.True (symbol.TryGetEnumFields (out var fields, out var diagnostics));
+		Assert.NotNull (fields);
+		Assert.Empty (fields);
+		Assert.Null (diagnostics);
```

Comment:
It should be an error to use invalid field names in the Field attribute, from what I read here it just seems like we silently skip over such fields?

Reply:
Yes, the validation is done by the analyzer, which will result in a compilation error.